esmtools/carbon.py

```diff
@@ -125,3 +123,180 @@ def potential_pco2(t_insitu, pco2_insitu):
     t_sfc = t_insitu.isel(depth=0)
     pco2_potential = pco2_insitu * (1 + 0.0423 * (t_sfc - t_insitu))
     return pco2_potential
+
+
+def spco2_sensitivity(ds):
+    """Generate sensitivities in spco2 for changes in other variables.
+
+    * Lovenduski, Nicole S., Nicolas Gruber, Scott C. Doney, and Ivan D. Lima.
+        “Enhanced CO2 Outgassing in the Southern Ocean from a Positive Phase of
+        the Southern Annular Mode.” Global Biogeochemical Cycles 21, no. 2
+        (2007). https://doi.org/10/fpv2wt.
+    * Gruber and Sarmiento, 2005
+
+    Args:
+        ds (xr.Dataset): containing cmorized variables:
+                            spco2 [ppm]: pCO2,ocean at ocean surface
+                            talkos[mmol m-3]: Alkalinity at ocean surface
+                            dissicos[mmol m-3]: DIC at ocean surface
+                            tos [C] : temperature at ocean surface
+                            sos [psu] : salinity at ocean surface
+
+    Returns:
+        sensitivity (xr.Dataset):
+
+    """
+    def _check_variables(ds):
+        requiredVars = ['spco2', 'tos', 'sos', 'talkos', 'dissicos']
+        if not all(i in ds.data_vars for i in requiredVars):
+            missingVars = [i for i in requiredVars if i not in ds.data_vars]
+            raise ValueError(f"""Missing variables needed for calculation:
+            {missingVars}""")
+
+    _check_variables(ds)
+    # Sensitivities are based on the time-mean for each field. This computes
+    # sensitivities at each grid cell.
+    # TODO: Add keyword for sliding mean, as in N year chunks of time to
+    # account for trends.
+    DIC = ds['dissicos']
+    ALK = ds['talkos']
+    SALT = ds['sos']
+    pCO2 = ds['spco2']
+
+    buffer_factor = dict()
+    buffer_factor['ALK'] = -ALK**2 / ((2 * DIC - ALK) * (ALK - DIC))
+    buffer_factor['DIC'] = (3*ALK*DIC - 2*DIC**2) / \
+                           ((2 * DIC - ALK) * (ALK - DIC))
+    # Compute sensitivities
+    sensitivity = dict()
+    sensitivity['tos'] = 0.0423
+    sensitivity['sos'] = 1 / SALT
+    sensitivity['talkos'] = (1 / ALK) * buffer_factor['ALK']
+    sensitivity['dissicos'] = (1 / DIC) * buffer_factor['DIC']
+    sensitivity = xr.Dataset(sensitivity) * pCO2
+    return sensitivity
+
+
+# TODO: adapt for CESM and MPI output.
```

Comment:
probably cmorized output is just fine. and folks should do the conversion to cmorized units themselves

Reply:
Agreed.